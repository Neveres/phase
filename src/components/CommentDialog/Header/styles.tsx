import { css } from '@emotion/react'

export const headerContainer = css`
  display: flex;
  justify-content: space-between;
  .ant-switch {
    margin-right: 10px;
  }
`
