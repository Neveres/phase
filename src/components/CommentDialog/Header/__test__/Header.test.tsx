import React from 'react'
import { create } from 'react-test-renderer'
import Header from '../Header'

describe('Header', () => {
  test('should render well uuid existed', () => {
    const uuid = 'uuid'
    expect(
      create(
        <Header
          uuid={uuid}
          isResolved={false}
          onDelete={jest.fn()}
          switchRef={null as any}
        />,
      ).toJSON(),
    ).toMatchSnapshot()

    expect(
      create(
        <Header
          uuid={uuid}
          isResolved={true}
          onDelete={jest.fn()}
          switchRef={null as any}
        />,
      ).toJSON(),
    ).toMatchSnapshot()
  })

  test('should render well uuid not existed', () => {
    const uuid = ''
    expect(
      create(
        <Header
          uuid={uuid}
          isResolved={false}
          onDelete={jest.fn()}
          switchRef={null as any}
        />,
      ).toJSON(),
    ).toMatchSnapshot()

    expect(
      create(
        <Header
          uuid={uuid}
          isResolved={true}
          onDelete={jest.fn()}
          switchRef={null as any}
        />,
      ).toJSON(),
    ).toMatchSnapshot()
  })
})
