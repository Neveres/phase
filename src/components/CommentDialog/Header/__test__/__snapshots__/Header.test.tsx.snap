// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render well uuid existed 1`] = `
<div
  className="css-1uj4mpx-Header"
>
  <div>
    <button
      aria-checked={false}
      className="ant-switch"
      data-testid="resolved-switch"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="switch"
      type="button"
    >
      <div
        className="ant-switch-handle"
      />
      <span
        className="ant-switch-inner"
      />
    </button>
    <span>
      Resolved
    </span>
  </div>
  <div
    className="ant-dropdown-trigger"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    Options
  </div>
</div>
`;

exports[`Header should render well uuid existed 2`] = `
<div
  className="css-1uj4mpx-Header"
>
  <div>
    <button
      aria-checked={true}
      className="ant-switch ant-switch-checked"
      data-testid="resolved-switch"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="switch"
      type="button"
    >
      <div
        className="ant-switch-handle"
      />
      <span
        className="ant-switch-inner"
      />
    </button>
    <span>
      Resolved
    </span>
  </div>
  <div
    className="ant-dropdown-trigger"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    Options
  </div>
</div>
`;

exports[`Header should render well uuid not existed 1`] = `null`;

exports[`Header should render well uuid not existed 2`] = `null`;
